csharp/CMakeLists.txt

```diff
@@ -265,18 +235,36 @@ add_custom_target(csharp_sdk ALL
 )
 
 if( BUILD_NUGET_PACKAGE )
+  if(NOT CMAKE_CL_64)
+    message(FATAL_ERROR "Nuget package cannot be built using the 64 bit generator")
+  endif()
 
   if (MSVC)
+    include(ExternalProject)
+    
+    ExternalProject_Add(OS-32
+      SOURCE_DIR ${PROJECT_SOURCE_DIR}
+      BINARY_DIR ${PROJECT_BINARY_DIR}/OS-32
+      BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config Release
+      CMAKE_GENERATOR
+        "Visual Studio 15 2017"
```

Comment:
This shouldn't be harcoded to MSVC 2017. I could be on a machine with only MSVC 2019 too.

Reply:
Yeah something better here for sure. We also would like to use Ninja instead of VS. But it isn't as simple as just using the generator of the parent project because we want the 32 bit variant. For VS the 32 / 64 bit variants are handled by different generator targets, although that is not true for all types of generators. @tijcolem was researching ninja.

Reply:
This thread talks about this exact issue. 
https://gitlab.kitware.com/cmake/cmake/-/issues/17536  

Basically, we need to pass the Visual Studio x86 32 bit envs to cmake so ninja can pick it up and set the correct config.  I _believe_ that can be accomplished by creating a setup script that calls the VS tools vcvars32.bat script. We are using GitBash for our environment so this would also need to translated to a bash script.  

So something like 

`CONFIGURE_COMMAND /path/to/vcvars_x86.sh`

or maybe `CONFIGURE_COMMAND "cmd /C /path/to/vcvars32.bat" 
`
I think getting both 64 and 32 bit to use ninja generator should be the goal, but I don't want to hold the PR up to implement this just for the 32 bit build nugget package. If we can get ninja to build the 64 bit and then use Visual Studio generator for the 32 bit, my vote would be to proceed and then do more testing with using ninja for the external project.  I'm testing that setup now.   Running into the $CONFIGURATION for the nugget template as mentioned above.